### ISSUE TITLE
Build system should not assume everything is vendored and bundled

### ISSUE THREAD

**Issue by jmoraleda** [Murmele/Gittyup#581]
I think this project should update build system so that it does not assume everything is vendored and bundled. It should support normal methods (such as pkg-config) to detect the required dependencies when they are provided by the system and build using them and install using normal methods.

Preliminary work has been done in #191 but more work is needed.

This is a feature request not a bug report.

### Rationale

This feature would make it much easier for either the maintainers of the project, or downstream distributions, to package and distribute this project in a variety of binary formats.

To my knowledge this feature was first suggested a year ago in a comment https://github.com/Murmele/Gittyup/issues/198#issuecomment-1161543223 but I think the feature should be discussed by itself regardless of which binary formats the project chooses to distribute as no one binary format will be universally acceptable to everyone, including #579.

**Comment by Murmele** [comment]
@jmoraleda I agree with you. One big topic is libgit2 where we are using currently a custom branch and not the upstream branch (https://github.com/Murmele/Gittyup/pull/153 ongoing work and patches submitted as PR upstream). I think for the rest only the options must be changed from OFF to ON as it is done in the AUR build (https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=gittyup)

**Comment by mwerle** [comment]
Just came here to post the same ticket, glad to see it's already being discussed and that upstream is amenable to this.

As regards patches to libgit2 - that will take a LONG time to filter through to released distros unless they are critical security vulnerabilities. Are the custom changes to libgit2 really that necessary or just "nice to have"?

**Comment by Murmele** [comment]
> Just came here to post the same ticket, glad to see it's already being discussed and that upstream is amenable to this.
> 
> As regards patches to libgit2 - that will take a LONG time to filter through to released distros unless they are critical security vulnerabilities. Are the custom changes to libgit2 really that necessary or just "nice to have"?

Some of the libgit2 are neccessary, some are already merged. For some I did not understand why they are needed.
A summary can be found here: https://github.com/Murmele/Gittyup/pull/153

**Comment by mmuman** [comment]
I second that: trying to port things to Haiku in a VM and always having to move things around to make space for gigabytes of dependencies as submodules when we already have them packaged is… annoying. At least there are less hardcoded ones than gitahead already.

**Comment by AHSauge** [comment]
I'll be a bit honest here and say that it's quite frustrating to get a build environment up and running on Windows (it's a general issue for me with Windows development), and I assume this has contributed to the current situation with submodules and config variables.

That said though, there are some pretty good ways to streamline this. Would there be any issues using conan for example? There is [cmake integration](https://github.com/conan-io/cmake-conan) such that `find_package` is rerouted to conan if the system doesn't provide a given dependency, which in turn will install or build the dependencies. There's some extra fluff that needs to be added to the setup call for cmake (which implicitly makes conan optional), but otherwise it's rather invisible. For platforms without conan you're a bit on your own of course, but I reckon people in that situation have a good enough understanding of how to setup the environment that providing the dependencies through the system isn't an issue.
